Nurlan database week2

nurlanimamali95 opened the pull request.

nurlanimamali95 commented:
I would like to mention that somehow the commits from week1 also added here. Please ignor them. So the commits for week two are 'connection created', 'done with exe1', 'done with exe3', 'smalll changes for exe 2 and finsihed exe3'.

jrnhofman commented on Week2/database/exe2.js:
```diff
@@ -0,0 +1,109 @@
+const mysql = require("mysql");
```
Hey! For this exercise please keep in mind what kind of relationship you want to create between authors and papers. What kind of relationship would you want to create? And in that case, what kind of setup do you need to use here? (look in the lecture notes if you don't know, if you can't find let me know and I can help you).

jrnhofman commented on Week2/database/exe2.js:
```diff
@@ -0,0 +1,109 @@
+    console.log("Added author_id column to research_Papers table successfully");
+
+    const insertAuthorsDataQuery = `
+      INSERT INTO authors (author_id, author_name, university, date_of_birth, h_index, gender)
```
This doesn't run for me, are you sure you are inserting the correct values?

jrnhofman commented on Week2/database/exe3.js:
```diff
@@ -0,0 +1,84 @@
+const mysql = require("mysql");
```
This content needs to change pending changes to exercise 2.

jrnhofman replied:
Also this is exercise 4 right? Where is exercise 3? :)

nurlanimamali95 commented:
Hello Jeroen, Thanks for your feedback. I made important changes in my exercises. Before i was doing the exe using API. However it was not necessary. I added new changes without creating API. Added async await and also exe3. Thanks in advance

jrnhofman commented on Week2/database/exe4.js:
```diff
@@ -0,0 +1,80 @@
+
+const sumOfPapersUniversity = async () => {
+  const query4 = `
+    SELECT authors.university AS University, SUM (research_papers.paper_title) AS SumOfPapers
```
You are taking the SUM of a string column, can you do that?

nurlanimamali95 replied:
Thanks for your feedback. You are definitely right. I can not use SUM because of the string values. I made necessary changes, please let me know if anything else  I should consider

jrnhofman commented on Week2/database/exe4.js:
```diff
@@ -0,0 +1,81 @@
+
+const sumOfPapersUniversity = async () => {
+  const query4 = `
+    SELECT authors.university AS University, COUNT(*) AS SumOfPapers
```
You will need to use COUNT(DISTINCT(...)) here :)

nurlanimamali95 replied:
Thanks. Updated :)

jrnhofman commented:
In general, I would encourage you to look at the example data your create, write your queries and then check if the queries not only run, but also produce the result that answers the question. What I mean to say is that for each of those questions you know what the answer should be (since you created the test data) so you can check if your code actually gives the same result, if not then you can figure out if you maybe made a logical mistake (not a coding mistake, but a mistake in the logic of your query) :).
